File: smart_plant/src/main/java/com/smart_plant/smart_plant/entity/GrowthStage.java
package com.smart_plant.smart_plant.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 作物生长期实体，对应 growth_stage 表。
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class GrowthStage {

    /** 阶段ID */
    private Long id;

    /** 作物ID */
    private Long cropId;

    /** 作物名称，列表展示字段 */
    private String cropName;

    /** 阶段名称 */
    private String stageName;

    /** 阶段编码 */
    private String stageCode;

    /** 阶段顺序 */
    private Integer stageOrder;

    /** 开始天数 */
    private Integer startDay;

    /** 结束天数 */
    private Integer endDay;

    /** 持续天数 */
    private Integer duration;

    /** 建议光照时长（小时/天） */
    private BigDecimal lightHours;

    /** 最低适宜温度 */
    private BigDecimal tempMin;

    /** 最高适宜温度 */
    private BigDecimal tempMax;

    /** 最低适宜湿度 */
    private BigDecimal humidityMin;

    /** 最高适宜湿度 */
    private BigDecimal humidityMax;

    /** 最低适宜PH */
    private BigDecimal phMin;

    /** 最高适宜PH */
    private BigDecimal phMax;

    /** 最低适宜EC值 */
    private BigDecimal ecMin;

    /** 最高适宜EC值 */
    private BigDecimal ecMax;

    /** 建议浇水间隔天数 */
    private Integer waterIntervalDays;

    /** 建议施肥间隔天数 */
    private Integer fertilizerIntervalDays;

    /** 管理建议 */
    private String managementAdvice;

    /** 状态：1启用 0禁用 */
    private Integer status;

    /** 备注 */
    private String remark;

    /** 创建时间 */
    private LocalDateTime createTime;

    /** 更新时间 */
    private LocalDateTime updateTime;
}

File: smart_farm/src/api/growthStage.ts
import { get, http, post } from '@/utils/request'
import type { PageInfo } from '@/api/user'

/** 作物生长期接口封装，对应 smart_plant 后端 /growth-stage 系列接口。 */
export interface GrowthStage {
  id?: number
  cropId: number
  cropName?: string
  stageName: string
  stageCode?: string
  stageOrder: number
  startDay?: number
  endDay?: number
  duration?: number
  lightHours?: number
  tempMin?: number
  tempMax?: number
  humidityMin?: number
  humidityMax?: number
  phMin?: number
  phMax?: number
  ecMin?: number
  ecMax?: number
  waterIntervalDays?: number
  fertilizerIntervalDays?: number
  managementAdvice?: string
  status?: number
  remark?: string
  createTime?: string
  updateTime?: string
}

export interface GrowthStagePayload {
  cropId: number
  stageName: string
  stageCode?: string
  stageOrder: number
  startDay?: number
  endDay?: number
  duration?: number
  lightHours?: number
  tempMin?: number
  tempMax?: number
  humidityMin?: number
  humidityMax?: number
  phMin?: number
  phMax?: number
  ecMin?: number
  ecMax?: number
  waterIntervalDays?: number
  fertilizerIntervalDays?: number
  managementAdvice?: string
  status?: number
  remark?: string
}

export interface UpdateGrowthStagePayload extends Partial<GrowthStagePayload> {
  id: number
}

export interface GrowthStageListParams {
  cropId?: number
  stageName?: string
  status?: number
  pageNum: number
  pageSize: number
}

interface ApiResult<T> {
  code: number
  message: string
  data: T
}

export const listGrowthStages = (params: GrowthStageListParams) => {
  return get<ApiResult<PageInfo<GrowthStage>>>(
    '/growth-stage/list',
    params as unknown as Record<string, unknown>,
  )
}

export const getGrowthStageById = (id: number) => {
  return get<ApiResult<GrowthStage>>(`/growth-stage/${id}`)
}

export const addGrowthStage = (data: GrowthStagePayload) => {
  return post<ApiResult<GrowthStage>>(
    '/growth-stage/add',
    data as unknown as Record<string, unknown>,
  )
}

export const updateGrowthStage = (data: UpdateGrowthStagePayload) => {
  return http<ApiResult<GrowthStage>>({ url: '/growth-stage', method: 'put', data })
}

export const deleteGrowthStage = (id: number) => {
  return http<ApiResult<void>>({ url: `/growth-stage/${id}`, method: 'delete' })
}

export const batchDeleteGrowthStages = (ids: number[]) => {
  return http<ApiResult<number>>({ url: '/growth-stage/batch', method: 'delete', data: ids })
}

export const updateGrowthStageStatus = (id: number, status: number) => {
  return http<ApiResult<void>>({
    url: `/growth-stage/${id}/status`,
    method: 'put',
    params: { status },
  })
}
